src/main/java/org/sefglobal/scholarx/service/MentorService.java

```diff
@@ -210,4 +210,57 @@ public Mentee getLoggedInMentee(long mentorId, long profileId)
         }
         return optionalMentee.get();
     }
+
+    /**
+     * Update {@link EnrolmentState} of Approved {@link Mentee} objects of a user as Rejected
+     * to keep only one approved {@link Mentee}
+     *
+     * @param mentorId  which is the {@link Mentor} object's id of the {@link Mentee}
+     * @param profileId which is the id of the {@link Profile}
+     * @return the updated {@link Mentee}
+     *
+     * @throws ResourceNotFoundException is thrown if the {@link Mentor} doesn't exist
+     * @throws NoContentException        is thrown if user doesn't have any approved {@link Mentee} objects
+     * @throws BadRequestException       is thrown if the {@link Mentee} is not approved
+     */
+    public Mentee confirmMentor(long mentorId, long profileId)
+            throws ResourceNotFoundException, NoContentException, BadRequestException {
```

Comment:
Hi nice job with the logic but I have another suggestion to achieve the functionality,
1. Do the normal validations
2. Get the confirmed mentee application by: `findAllByProfileIdAndMentorId`(if there is a findOne method use that)
3. If we have an application: change the state and save the data back.
4. Write a native `@Modifying update` query in the repository to update all the other application rows of this mentee to removed state.
5. Return the changed mentee application

WDYT?

Reply:
Great idea. This may be much efficient than the previous one. I'll try to implement that.

Reply:
@YohanAvishke I updated the PR with this change. But I'm not sure whether it's the way you suggested. I just wrote a repository method to reject mentees. Could you please review it